projects/plugins/jetpack/_inc/client/my-plan/my-plan-partner-coupon/index.jsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getRedirectUrl } from '@automattic/jetpack-components';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import MyPlanBanner from '../my-plan-banner';
+import { getPartnerCoupon } from 'state/initial-state';
+
+const MyPlanPartnerCoupon = ( { partnerCoupon, siteRawUrl } ) => {
+	if ( 'object' !== typeof partnerCoupon ) {
+		return null;
+	}
+
+	const redeemButton = (
+		<Button
+			primary
+			href={ getRedirectUrl( 'jetpack-plugin-partner-coupon-checkout', {
+				path: partnerCoupon.product.slug,
+				site: siteRawUrl,
+				query: `coupon=${ partnerCoupon.coupon_code }`,
+			} ) }
+		>
+			{ __( 'Redeem', 'jetpack' ) }
+		</Button>
+	);
+
+	return (
+		<MyPlanBanner
+			productSlug={ partnerCoupon.product.slug }
+			action={ redeemButton }
+			title={ sprintf(
+				/* translators: %s: Jetpack product or plan name. */
+				__( 'Get %s free for one year!', 'jetpack' ),
+				partnerCoupon.product.title
+			) }
+			tagLine={ sprintf(
+				/* translators: %1$s: the name of a Jetpack partner, %2$s: the name of a Jetpack product or plan. */
+				__(
+					'Redeem your %1$s coupon to get started with %2$s for free the first year!',
```

Comment:
Are all coupons be 100% off for a yearly purchase? If that may change in the future, should we add a warning somewhere that this is currently the only supported type of coupon?

Reply:
We're going for a minimal implementation as it's not expected future for more types of coupons to come into play in the immediate.
Nevertheless, we're tracking this potential future work in 1201265858818217-as-1201728015534302.